clang/include/clang/Sema/Sema.h

```diff
@@ -12098,11 +12132,25 @@ class Sema final {
   /// before incrementing, so you can emit an error.
   bool PopForceCUDAHostDevice();
 
+  class DeviceDeferredDiagnostic {
+  public:
+    DeviceDeferredDiagnostic(SourceLocation SL, const PartialDiagnostic &PD,
+                             DeviceDiagnosticReason R)
+        : Diagnostic(SL, PD), Reason(R) {}
+
+    PartialDiagnosticAt &getDiag() { return Diagnostic; }
+    DeviceDiagnosticReason getReason() { return Reason; }
```

Comment:
```suggestion
    PartialDiagnosticAt &getDiag() const { return Diagnostic; }
    DeviceDiagnosticReason getReason() const { return Reason; }
```

Reply:
Done. `getDiag` cannot be made const since we change `PartialDiagnosticAt` through this ref.